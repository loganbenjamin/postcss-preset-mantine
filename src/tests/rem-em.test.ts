const { testTransform } = require('./utils');

const baseInput = `
.demo {
  font-size: rem(16px);
  height: em(40px);
}
`;

const mediaInput = `
@media (min-width: em(320px)) {
  font-size: rem(32px);
}
@media (min-height: rem(240px)) {
  font-size: em(16px);
}
`;

const remEmInsideClassNameInput = `
.button-rem {
  color: red;
}
.button-em {
  color: red;
}
`;

const remEmInsideFunctionInput = `
.button1 {
  color: theorem(32px);
}
.button2 {
  color: poem(48px);
}
`;

describe('rem-em', () => {
  it('should work', async () => {
    const res = await testTransform(baseInput);
    expect(res.css).toMatchSnapshot();
  });

  it('works inside media query', async () => {
    const res = await testTransform(mediaInput);
    expect(res.css).toMatchSnapshot();
  });

  it('works when rem/em is inside the class name', async () => {
    const res = await testTransform(remEmInsideClassNameInput);
    expect(res.css).toMatchSnapshot();
  });

  it('doesnt process the function when name doesnt match exactly', async () => {
    const res = await testTransform(remEmInsideFunctionInput);
    expect(res.css).toMatchSnapshot();
  });
});
