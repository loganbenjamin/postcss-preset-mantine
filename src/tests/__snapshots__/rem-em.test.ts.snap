// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rem-em doesnt process the function when name doesnt match exactly 1`] = `
"
.button1 {
  color: theorem(32px);
}
.button2 {
  color: poem(48px);
}
"
`;

exports[`rem-em should work 1`] = `
"
.demo {
  font-size: calc(1rem * var(--mantine-scale));
  height: 2.5em;
}
"
`;

exports[`rem-em works inside media query 1`] = `
"
@media (min-width: 20em) {
  font-size: calc(2rem * var(--mantine-scale));
}
@media (min-height: 15rem) {
  font-size: 1em;
}
"
`;

exports[`rem-em works when rem/em is inside the class name 1`] = `
"
.button-rem {
  color: red;
}
.button-em {
  color: red;
}
"
`;
